Admin/AbstractAdmin.php

```diff
@@ -2782,37 +2782,65 @@ public function configureActionButtons($action, $object = null)
     {
         $list = array();
 
-        if (in_array($action, array('tree', 'show', 'edit', 'delete', 'list', 'batch'))) {
+        if (
+            in_array($action, array('tree', 'show', 'edit', 'delete', 'list', 'batch'))
+            && $this->hasAccess('create')
+        ) {
             $list['create'] = array(
                 'template' => 'SonataAdminBundle:Button:create_button.html.twig',
             );
         }
 
-        if (in_array($action, array('show', 'delete', 'acl', 'history')) && $object) {
+        if (
+            in_array($action, array('show', 'delete', 'acl', 'history'))
+            && $object
+            && $this->id($object)
+            && $this->hasAccess('edit', $object)
```

Comment:
Should be `private` at first, if it turns out to be useful we'll make it `final public`